# common.c -- load both lowercase and uppercase pak variants
#
# The original engine probes only pak0.pak, pak1.pak, ... in lowercase. On
# case-sensitive filesystems this misses valid uppercase PAK0.PAK / PAK1.PAK
# files, and both forms can coexist with different contents. Load both
# variants for each pak index before stopping the search.
#
--- common.c
+++ common.c
@@ -1708,14 +1708,33 @@
 //
 	for (i=0 ; ; i++)
 	{
+		qboolean	loaded;
+
+		loaded = false;
 		sprintf (pakfile, "%s/pak%i.pak", dir, i);
 		pak = COM_LoadPackFile (pakfile);
-		if (!pak)
+		if (pak)
+		{
+			search = Hunk_Alloc (sizeof(searchpath_t));
+			search->pack = pak;
+			search->next = com_searchpaths;
+			com_searchpaths = search;
+			loaded = true;
+		}
+
+		sprintf (pakfile, "%s/PAK%i.PAK", dir, i);
+		pak = COM_LoadPackFile (pakfile);
+		if (pak)
+		{
+			search = Hunk_Alloc (sizeof(searchpath_t));
+			search->pack = pak;
+			search->next = com_searchpaths;
+			com_searchpaths = search;
+			loaded = true;
+		}
+
+		if (!loaded)
 			break;
-		search = Hunk_Alloc (sizeof(searchpath_t));
-		search->pack = pak;
-		search->next = com_searchpaths;
-		com_searchpaths = search;               
 	}
 
 //
